util/input.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <iostream>
+
+/**
+ * A file which defines basic functionality for blocking keyboard input, without
+ * needing to press <enter>.  This isn't necessarily the most optimal solution,
+ * but this is a basic starting point for the purpose of input for the examples.
+ */
+
+/// Macros needed for getting keyboard input without waiting user needing to
+/// press <enter>
+#if defined(_WIN32) || defined(_WIN64) // On Windows
+    #include <conio.h>
+    #define hebi_getchar_init()
+    #define hebi_getchar() _getch()
+    #define hebi_getchar_cleanup()
+#elif defined(__APPLE__) || defined(TARGET_OS_OSX) // On OSX
+    #define hebi_getchar_init() system ("/bin/stty raw")
+    #define hebi_getchar() getchar()
+    #define hebi_getchar_cleanup() system ("/bin/stty -raw")
+#else // On Linux
+    #include <stdlib.h>
+    #define hebi_getchar_init() system ("/bin/stty raw")
+    #define hebi_getchar() getchar()
+    #define hebi_getchar_cleanup() system ("/bin/stty -raw")
+#endif
+
+namespace hebi {
+namespace util {
+
+class Input
+{
+public:
+  static int getChar()
+  {
+    init();
```

Comment:
oh no..

Reply:
This forces a fork on every single call (`system("/bin/stty");`, where system [invokes fork](https://code.woboq.org/userspace/glibc/sysdeps/posix/system.c.html)). Moving `init();` into the constructor and having `getChar()` look like

```c++
static int getChar() {
  static Keyboard kb;
  return hebi_getchar();
}
```

ensures that the `init();` is called only once, and ensures that `init();` isn't actually ever called until the user called `Keyboard::getChar()`

Reply:
lol I didn't actually read the code apparently. Forget that I said anything